apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -119,35 +119,35 @@ defmodule OMG.Watcher.BlockGetter do
       ) do
     with {:ok, _} <- Core.chain_ok(state),
          tx_exec_results = for(tx <- transactions, do: OMG.State.exec(tx, :ignore)),
-         {:ok, state} <- Core.validate_executions(tx_exec_results, to_apply, state) do
+         {:ok, state2} <- Core.validate_executions(tx_exec_results, to_apply, state) do
```

Comment:
What's the purpose of these changes - imho it hurts readability. 
If someone needs to add other operation on state in between it requires to renumbering all others bellow? 
Also it increases the chance for choosing the wrong `stateN` by mistake.
Opting for revert.

Reply:
The reason was because state changes were spread around in a concise manner. I've made further improvements so that we're piping actions. Take a look.